File: backend/notifications/src/main/java/com/oxygensend/notifications/context/rest/MessagePayload.java
package com.oxygensend.notifications.context.rest;


import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;

public record MessagePayload<C>(@Valid
                         C content,
                         String login,
                         @NotEmpty
                         String serviceID,
                         String createdAt) {
}

File: backend/notifications/src/main/kotlin/com/oxygensend/notifications/context/rest/MessagePayload.kt
package com.oxygensend.notifications.context.rest

import jakarta.validation.Valid
import jakarta.validation.constraints.NotEmpty
import java.time.LocalDateTime

data class MessagePayload<C>(
    val content: @Valid C,
    val login: String?,
    val serviceId: @NotEmpty String,
    val requestId: String?,
    val createdAt: LocalDateTime?
)
